conda.recipe/meta.yaml

```diff
@@ -26,6 +26,7 @@ requirements:
     {% for dep in sdata.get('install_requires',{}) %}
     - {{ dep }}
     {% endfor %}
+    - libarrow-flight
```

Comment:
Pyarrow on conda-forge has begun splitting its library into components to reduce its size. I need to check if this is possible on the default channel.

Reply:
Got my answer:

![image](https://github.com/intake/intake-dremio/assets/19758978/f84e3d19-c305-41a3-bb7b-4c230cbfb8d3)

Reply:
For now, I have just added an upper pin for pyarrow on the conda recipe. I can't think of a better way to do it.

Reply:
Can you explain? I don't follow why the upper pin is needed? Is the problem that we use conda-forge for the build so we would have to add the additional flight dependency? If so let's not build against conda-forge for now. I'll simply update the conda-forge recipe to request the pyarrow flight pin.

Reply:
Pyarrow on conda-forge started to split `pyarrow` package into multiple sub-packages with version 16, this is to reduce the size of the library. If we don't have `libarrow-flight` installed when using conda-forge, the `intake_dremio` import will fail. 

I tried to add it as a dependency, but then the CI, which uses the default channel, would not solve as it doesn't have a `libarrow-flight` package.

So we are between a rock and a hard place. So I chose to take the easiest way as far as I can see, which is to upper pin to 15, which was before conda-forge split up the `pyarrow` package.  When we do an official release, we can update the respective recipe for both conda-forge and default, but this package should work on both of them.

Reply:
I plan to do an official release shortly. Can we not simply stop using conda-forge in our CI and then I will add the `libarrow-flight` dependency specifically on the conda-forge recipe?

Reply:
Our CI does not use conda-forge, but I do locally. 

If we do what you suggest, the `pyviz + conda-forge` will fail when importing `intake-dremio`.

Reply:
Hmm, good point, thanks for clarifying. That said an upper pin on pyarrow makes this all pretty useless, the point of the release is to support recent versions of pyarrow. In the end the pyviz channel isn't super important so we may just have to document this issue in the README and suggest users either install both packages from conda-forge OR explicitly install `libarrow-flight`.

Reply:
As long as we release on defaults (no pin) + conda-forge (no pin, add `libarrow-flight`), it should support new versions fine. The pin on the package on pyviz will likely be bypassed. And definitely would if a higher pin is specified on pyarrow.

Reply:
I'm not particularly convinced that is true, pyviz channel is generally added with highest priority so unless you pin recent pyarrow I think it will pull down the pyviz channel version and then pull an older pyarrow version.